File: backend/app/core/upgrades/stat_modifiers/damage_5.py
from ..upgrade_implementation import UpgradeImplementation, UpgradeAppliers


class Damage5(UpgradeImplementation):
    id = 'damage_5'
    name = 'Devastation'
    description = '+7.5% damage.'
    rarity = 'legendary'
    type = 'stat_modifier'
    cost = 40

    def can_apply(self, state: dict) -> bool:
        return True

    def apply(self, appliers: UpgradeAppliers) -> None:
        value = 0.075
        appliers.modifiers.add_multiplier('attack', 'damage', value)

    def validate(self, recorded_stats: dict) -> bool:
        return True

File: frontend/src/game/upgrades/stat_modifiers/damage_5.ts
import { Upgrade, type UpgradeDef } from '../Upgrade'
import { UpgradeTargetID, UpgradeStatID, RarityID, UpgradeTypeID } from '../../data/ID'

export const Damage5Def: UpgradeDef = {
  id: "damage_5",
  name: "Devastation",
  description: "+7.5% damage.",
  rarity: RarityID.Legendary,
  upgradeType: UpgradeTypeID.StatModifier,
  cost: 40,
  targetClass: UpgradeTargetID.Attack,
  fieldInTargetClass: UpgradeStatID.Damage,
  value: 0.075,
  isMultiplier: true,
  stackable: true,
  maxStacks: 99999,
}

export class Damage5 extends Upgrade {}
